shell/browser/file_system_access/file_system_access_permission_context.cc

```diff
@@ -610,16 +610,18 @@ void FileSystemAccessPermissionContext::PerformAfterWriteChecks(
 }
 
 void FileSystemAccessPermissionContext::RunRestrictedPathCallback(
+    const base::FilePath& file_path,
     SensitiveEntryResult result) {
-  if (callback_)
-    std::move(callback_).Run(result);
+  if (base::Contains(callback_map_, file_path))
+    std::move(callback_map_[file_path]).Run(result);
```

Comment:
the `std::move()` consumes the callback, so we want to remove it from the map when we use it.

```suggestion
  if (auto val = callback_map_.extract(file_path))
    std::move(val.mapped()).Run(result);
```